scripts/bfb-install

```diff
@@ -371,18 +371,47 @@ push_boot_stream()
   fi
 }
 
+check_nic_mode()
+{
+  local str
+
+  [ "$mode" != "local" ] && return
+
+  # Get PCIE BDF
+  str=`cat ${rshim_node}/misc | grep DEV_NAME | awk '{print $2}'`
+  str=${str:5}
+  str=${str/.*/.0}
+  pcie_bdf="$str"
+
+  # Check NIC mode
+  str=`mlxconfig -d "$str" q INTERNAL_CPU_OFFLOAD_ENGINE 2>/dev/null | grep INTERNAL_CPU_OFFLOAD_ENGINE | grep 'DISABLED(1)'`
+  str=`echo "${str}" | awk '{print $2}'`
+  if [ ."$str" = ."DISABLED(1)" ]; then
+    is_nic_mode=1
+  else
+    is_nic_mode=0
+  fi
+}
+
 # Wait for RSHIM to finish updating by monitoring keywords in the RSHIM log
 #
 # Global variables used:
 #   $mode, $rshim_node, $remote_mode
 wait_for_update_to_finish()
 {
-  local filter="Reboot|finished|DPU is ready|Linux up"
+  local filter0 filter
```

Comment:
Better add a comment here on why we need two filters. Like "filter" is for installation completely finished, and "filter0" is for partially finished but OK to turn off CLEAR_ON_READ?

Reply:
None, added comment for it:
  # 'filter0' indicates bfb installation completion, thus CLEAR_ON_READ
  # can be disabled for next boot. 'filter' is related to specific mode
  # which might include extra configuration or booting, and is used as
  # the exit condition of the bfb-install script.